app/helpers/layout_helper.rb

```diff
@@ -193,28 +193,61 @@ def autocomplete_f(f, attr, options = {})
   end
 
   def field(f, attr, options = {})
-    error = f.object.errors[attr] if f && f.object.respond_to?(:errors)
+    table_field = options.delete(:table_field)
+    error       = f.object.errors[attr] if f && f.object.respond_to?(:errors)
     help_inline = help_inline(options.delete(:help_inline), error)
-    help_block  = content_tag(:span, options.delete(:help_block), :class => "help-block")
-    size_class = options.delete(:size) || "col-md-4"
-    content_tag(:div, :class=> "clearfix") do
-      content_tag :div, :class => "form-group #{error.empty? ? "" : 'has-error'}",
-                  :id          => options.delete(:control_group_id) do
-        required = options.delete(:required) # we don't want to use html5 required attr so we delete the option
-        required_mark = ' *' if required.nil? ? is_required?(f, attr) : required
-        label   = options[:label] == :none ? '' : options.delete(:label)
-        label ||= ((clazz = f.object.class).respond_to?(:gettext_translation_for_attribute_name) &&
-            s_(clazz.gettext_translation_for_attribute_name attr)) if f
-        label   = label.present? ? label_tag(attr, "#{label}#{required_mark}".html_safe, :class => "col-md-2 control-label") : ''
-        fullscreen = options[:fullscreen] ? fullscreen_button("$(this).prev().find('textarea')") : ""
-        label.html_safe +
-          content_tag(:div, :class => size_class) do
+    size_class  = options.delete(:size) || "col-md-4"
+    label = add_label(options, f, attr)
+
+    if !table_field
```

Comment:
Nitpick, use `unless` or preferably swap the two blocks of code around so this reads `if table_field`.  (There's a rubocop TODO to enforce this.)